daft/daft/__init__.pyi

```diff
@@ -1642,9 +1645,8 @@ class LogicalPlanBuilder:
         left_on: list[PyExpr],
         right_on: list[PyExpr],
         join_type: JoinType,
-        strategy: JoinStrategy | None = None,
-        join_prefix: str | None = None,
-        join_suffix: str | None = None,
+        join_strategy: JoinStrategy | None = None,
+        column_renaming_params: JoinOptions | None = None,
```

Comment:
personal preference: for the dataframe API, i actually like having it all flattened inside the `.join`. In python it seems pretty common to have methods with many kwargs, but less common to have methods with nested option classes.